experimental/lib/Transforms/Speculation/PlacementFinder.cpp

```diff
@@ -33,14 +33,6 @@ PlacementFinder::PlacementFinder(SpeculationPlacements &placements)
   assert(specPos.getOwner() && "Speculator position is undefined");
 }
 
-void PlacementFinder::clearPlacements() {
-  // Speculator and buffer positions are manually set
-  OpOperand &specPosition = placements.getSpeculatorPlacement();
-  llvm::DenseSet<OpOperand *> bufferPositions =
-      this->placements.getPlacements<handshake::BufferOp>();
-  this->placements = SpeculationPlacements(specPosition, bufferPositions);
-}
```

Comment:
`clearPlacement` used to discard all manual configurations except for `specPosition` and `bufferPositions`. The `placements` variable here is a reference from the `HandshakeSpeculation` pass, and this behavior of clearing the object owned by the pass was actually quite confusing (I got tripped up by it multiple times). It wipes unrelated manual configurations for no clear reason, so I've removed that behavior now.